src/agentscope/server/launcher.py

```diff
@@ -0,0 +1,464 @@
+# -*- coding: utf-8 -*-
+""" Server of distributed agent"""
+import os
+from multiprocessing import Process, Event, Pipe
+from multiprocessing.synchronize import Event as EventClass
+import socket
+import asyncio
+import signal
+import argparse
+from typing import Type, Optional
+from concurrent import futures
+from loguru import logger
+
+try:
+    import grpc
+    from agentscope.rpc.rpc_agent_pb2_grpc import (
+        add_RpcAgentServicer_to_server,
+    )
+except ImportError as import_error:
+    from agentscope.utils.tools import ImportErrorReporter
+
+    grpc = ImportErrorReporter(import_error, "distribute")
+    add_RpcAgentServicer_to_server = ImportErrorReporter(
+        import_error,
+        "distribute",
+    )
+
+import agentscope
+from agentscope.server.servicer import AgentServerServicer
+from agentscope.agents.agent import AgentBase
+from agentscope.utils.tools import _get_timestamp
+
+
+def setup_agent_server(
```

Comment:
The methods that are not planned to expose to developers should be prefixed by "_".